octoprint_octobullet/__init__.py

```diff
@@ -29,10 +33,106 @@ def __init__(self):
+
+		currentData["progress"]["printTimeLeftString"] = "No ETL yet"
+		currentData["progress"]["printTimeString"] = "Not started yet"
+		currentData["progress"]["ETA"] = "No ETA yet"
+		#Add additional data
+		try:
+			currentData["progress"]["printTimeLeftString"] = self._get_time_from_seconds(currentData["progress"]["printTimeLeft"])
+			currentData["progress"]["printTimeString"] = self._get_time_from_seconds(currentData["progress"]["printTime"])
+			currentData["progress"]["ETA"] = time.strftime(self._eta_strftime, time.localtime(time.time() + currentData["progress"]["printTimeLeft"]))
+		except Exception as e:
+			self._logger.warning("Caught an exception trying to parse data: {0}\n Error is: {1}\nTraceback:{2}".format(currentData,e,traceback.format_exc()))
+			
+		return currentData
+
+	def _get_time_from_seconds(self, seconds):
+		days =0
+		hours = 0
+		minutes = 0
+		if seconds >= 86400:
+			days = int(seconds / 86400)
+			secionds = seconds % 86400
+		if seconds >= 3600:
+			hours = int(seconds / 3600)
+			seconds = seconds % 3600
+		if seconds >= 60:
+			minutes = int(seconds / 60)
+			seconds = seconds % 60
+		return self._etl_format.format(**locals())
+	
+	#~~ PrintProgressPlugin
+	def on_print_progress(self, storage, path, progress):
+		if(progress%self._pc_interval==0):
+			self._quiet_time_sec = int(self._settings.get(["quiet_minutes"])) * 60
+			if self._quiet_time_sec == 0:
+				self._quiet_time_sec = 7* 24 * 3600 # use a week if we don't want messages
+			self._logger.info("Progress: {} {} {}".format(storage,path,progress))
+			try:
+				currentData = self._printer.get_current_data()
+				currentData = self._sanitize_current_data(currentData)
+			except Exception as e:
+				self._logger.info("Caught an exception {0}\nTraceback:{1}".format(e,traceback.format_exc()))
+
+			self._logger.info("Messasge: Remaining {} ({})".format(currentData["progress"]["printTimeLeftString"],currentData["progress"]["printTimeLeftOrigin"]))
+			self._logger.info("Messasge: Total Print time {}".format(currentData["progress"]["printTimeString"]))
+			self._logger.info("Messasge: Estimated Completion  {}".format(currentData["progress"]["ETA"]))
+			# first 3 messages, re-calculate interval (not first, analysis estimate can be inacurate)
+			self._message_count += 1
+			if self._message_count < 4 and self._message_count > 1:
+				total_job = currentData["progress"]["printTime"]+currentData["progress"]["printTimeLeft"]
+				self._pc_interval = int(100 * self._quiet_time_sec / total_job)
+				if self._pc_interval == 0:
+					self._pc_interval = 1
+				self._logger.info("Messasge: Set interval to {} percent count {} Total estimated {}".format(self._pc_interval, self._message_count,total_job))
+				
+			# Suppress message if print is nearly done or interval calculation is off
+			no_message = 0
+			if(currentData["progress"]["printTime"] < (self._quiet_time_sec * 0.5  + self._last_message)):
```

Comment:
Could you please explain what "(self._quiet_time_sec * 0.5  + self._last_message)" is supposed to accomplish here? As it is I have no idea what is going on here :/